### PR TITLE
NO-TICKET/configurable broadcast test

### PR DESCRIPTION
**Pull request by marc-casperlabs**
This PR allows tweaking the parameters of a built-in test from the commandline (via envvars) if desired, to avoid the considerable recompilation time.

### CODE REVIEW

File: node/src/testing/network.rs [hunk 1]
```diff
@@ -276,6 +276,23 @@ where
         &self.nodes
     }
 
+    /// Returns the internal map of nodes.
+    pub fn nodes_mut(&mut self) -> &mut HashMap<R::NodeId, Runner<ConditionCheckReactor<R>>> {
+        &mut self.nodes
+    }
```
Comment (Fraser999) [inline node/src/testing/network.rs:282]: We might as well just make the member variable public

File: node/src/components/network/tests_bulk_gossip.rs [hunk 2]
```diff
@@ -160,6 +185,10 @@ async fn send_large_message_across_network() {
 
     net.add_node_with_config(cfg, &mut rng).await.unwrap();
 
+    // Hack to get network component to connect. This gives the libp2p thread (which is independent
+    // of cranking) a little time to bind to the socket.
+    std::thread::sleep(std::time::Duration::from_secs(2));
```
Comment (Fraser999) [inline node/src/components/network/tests_bulk_gossip.rs:190]:
```suggestion
    thread::sleep(Duration::from_secs(2));
```

**Review by Fraser999 — APPROVED** [review]
Couple of nitpicks, but LGTM otherwise.

**Review by EdHastingsCasperLabs — APPROVED** [review]

File: node/src/testing/network.rs (hunk 1 above)
Comment (marc-casperlabs) [reply node/src/testing/network.rs:282]: I reserve the right to change the underlying data structure in the future :)

File: node/src/components/network/tests_bulk_gossip.rs (hunk 2 above)
Comment (marc-casperlabs) [reply node/src/components/network/tests_bulk_gossip.rs:190]: ✓

File: node/src/components/network/tests_bulk_gossip.rs [hunk 3]
```diff
@@ -127,7 +129,31 @@ impl Display for DummyPayload {
...
+        Ok(raw) => Some(
+            raw.parse()
+                .unwrap_or_else(|_| panic!("cannot parse envvar `{}`", name)),
+        ),
+        Err(env::VarError::NotPresent) => None,
+        Err(err) => {
+            panic!(err)
+        }
+    }
+}
```
Comment (goral09) [inline node/src/components/network/tests_bulk_gossip.rs:153]: This method is useful enough to have it in some `utils` package.

Comment (marc-casperlabs) [reply node/src/components/network/tests_bulk_gossip.rs]: Can do! :)

**Pull request merged**